### PR TITLE
Don't swallow test failures caused by POSIX signals

### PR DESCRIPTION
**Pull request by ramosbugs**
The `run-tests.js` script is used for running the Next.JS regression tests on GitHub Actions. It uses `child_process.spawn` to run each test in a child process, whose `exit` event [returns both a `code` and `signal`](https://nodejs.org/api/child_process.html#event-exit). Unfortunately, the current script only checks `code`, which swallows any test failures due to, e.g., Node.JS throwing a SIGSEGV. This has the potential to hide significant failures in the CI pipeline.

Before patch (succeeds despite SIGSEGV!):
```
$ node run-tests.js --timings --debug test/integration/amphtml/test/index.test.js
Running tests with concurrency: 2
Running tests: 
 test/integration/amphtml/test/index.test.js

Starting test/integration/amphtml/test/index.test.js retry 0/1
"next/jest" is currently experimental. https://nextjs.org/docs/messages/experimental-jest-transformer
info  - Using locally built binary of @next/swc
  console.log
    Running command "next build /Users/daramos/Documents/Projects/next.js/test/integration/amphtml/"

      at lib/next-test-utils.js:167:17

Finished test/integration/amphtml/test/index.test.js on retry 0/1 in 10.757s
$ echo $?
0
```

After patch (fails as expected):
```
$ node run-tests.js --timings --debug test/integration/amphtml/test/index.test.js
Running tests with concurrency: 2
Running tests: 
 test/integration/amphtml/test/index.test.js

Starting test/integration/amphtml/test/index.test.js retry 0/1
"next/jest" is currently experimental. https://nextjs.org/docs/messages/experimental-jest-transformer
  console.log
    Running command "next build /Users/daramos/Documents/Projects/next.js/test/integration/amphtml/"

      at lib/next-test-utils.js:167:17

CHILD EXITED WITH null
Cleaning test files at /Users/daramos/Documents/Projects/next.js/test/integration/amphtml/test
Starting test/integration/amphtml/test/index.test.js retry 1/1
"next/jest" is currently experimental. https://nextjs.org/docs/messages/experimental-jest-transformer
  console.log
    Running command "next build /Users/daramos/Documents/Projects/next.js/test/integration/amphtml/"

      at lib/next-test-utils.js:167:17

CHILD EXITED WITH null
test/integration/amphtml/test/index.test.js failed due to Error: failed with signal: SIGSEGV
test/integration/amphtml/test/index.test.js failed to pass within 1 retries
$ echo $?
1
```

Locally (on Mac OS 12.1 tested with both Node 16.13.1 and 14.18.2 on latest Next.JS `canary` branch), I'm running into segfaults due to https://github.com/nodejs/node/issues/35889 (see above), and I have a worrying suspicion that this same issue may be causing Next.JS tests to silently fail on CI too.

### CODE REVIEW

File: run-tests.js [hunk 1]
```diff
@@ -275,9 +275,9 @@ async function main() {
 
       children.add(child)
 
-      child.on('exit', async (code) => {
+      child.on('exit', async (code, signal) => {
         children.delete(child)
-        if (code) {
+        if (code || signal !== null) {
```
Comment (tjni) [inline run-tests.js:280]: What a great catch! I would suggest the condition be `if (code !== 0)`, which to me communicates the intent very slightly better.

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 7b3612a0fe6db8b322546c0be9d28e34334fbaee

**test/integration/legacy-pkg-gently/test/index.test.js** 
- Legacy Packages > should support `node-gently` packages

<details>
<summary>Expand output</summary>

  ● Legacy Packages › should support `node-gently` packages

    expect(received).toMatch(expected)

    Expected pattern: /hello world/i
    Received string:  "Internal Server Error"

      22 |       dir: appDir,
      23 |       dev: false,
    > 24 |       quiet: true,
         |                   ^
      25 |     })
      26 |
      27 |     server = await startApp(app)

      at Object.<anonymous> (integration/legacy-pkg-gently/test/index.test.js:24:21)

</details>

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| buildDuration | 22.2s | 22.2s | ⚠️ +15ms |
| buildDurationCached | 4.3s | 4.2s | -87ms |
| nodeModulesSize | 348 MB | 348 MB | ⚠️ +585 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.888 | 4.008 | ⚠️ +0.12 |
| / avg req/sec | 642.95 | 623.68 | ⚠️ -19.27 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.112 | 2.111 | 0 |
| /error-in-render avg req/sec | 1183.94 | 1184.04 | +0.1 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 30.3 kB | 30.3 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 74.1 kB | 74.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 326 B | 326 B | ✓ |
| dynamic-HASH.js gzip | 2.37 kB | 2.37 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 919 B | 919 B | ✓ |
| image-HASH.js gzip | 4.73 kB | 4.73 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.13 kB | 2.13 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 547 B | 547 B | ✓ |
| withRouter.html gzip | 528 B | 528 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| buildDuration | 24.4s | 24s | -404ms |
| buildDurationCached | 4.2s | 4.2s | ⚠️ +36ms |
| nodeModulesSize | 348 MB | 348 MB | ⚠️ +585 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 4.029 | 3.864 | -0.17 |
| / avg req/sec | 620.47 | 646.93 | +26.46 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.135 | 2.133 | 0 |
| /error-in-render avg req/sec | 1170.77 | 1172.02 | +1.25 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 30.4 kB | 30.4 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 74.3 kB | 74.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.36 kB | 2.36 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 906 B | 906 B | ✓ |
| image-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.19 kB | 2.19 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| _buildManifest.js gzip | 458 B | 458 B | ✓ |
| Overall change | 458 B | 458 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| index.html gzip | 530 B | 530 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
Commit: 459df00d4b569bbd99501ea57455884692a4fc34

**Review by ijjk — APPROVED** [review]
Good catch, thanks for the PR!

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| buildDuration | 19.6s | 19.7s | ⚠️ +65ms |
| buildDurationCached | 3.8s | 4s | ⚠️ +165ms |
| nodeModulesSize | 348 MB | 348 MB | ⚠️ +579 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.26 | 3.343 | ⚠️ +0.08 |
| / avg req/sec | 766.79 | 747.72 | ⚠️ -19.07 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.497 | 1.577 | ⚠️ +0.08 |
| /error-in-render avg req/sec | 1669.99 | 1585.42 | ⚠️ -84.57 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 30.3 kB | 30.3 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 74.1 kB | 74.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.37 kB | 1.37 kB | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 326 B | 326 B | ✓ |
| dynamic-HASH.js gzip | 2.37 kB | 2.37 kB | ✓ |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 919 B | 919 B | ✓ |
| image-HASH.js gzip | 4.73 kB | 4.73 kB | ✓ |
| index-HASH.js gzip | 263 B | 263 B | ✓ |
| link-HASH.js gzip | 2.13 kB | 2.13 kB | ✓ |
| routerDirect..HASH.js gzip | 321 B | 321 B | ✓ |
| script-HASH.js gzip | 383 B | 383 B | ✓ |
| withRouter-HASH.js gzip | 318 B | 318 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| _buildManifest.js gzip | 459 B | 459 B | ✓ |
| Overall change | 459 B | 459 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| index.html gzip | 533 B | 533 B | ✓ |
| link.html gzip | 547 B | 547 B | ✓ |
| withRouter.html gzip | 528 B | 528 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| buildDuration | 21.1s | 21.2s | ⚠️ +123ms |
| buildDurationCached | 3.8s | 3.8s | -50ms |
| nodeModulesSize | 348 MB | 348 MB | ⚠️ +579 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.356 | 3.402 | ⚠️ +0.05 |
| / avg req/sec | 744.89 | 734.76 | ⚠️ -10.13 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.51 | 1.58 | ⚠️ +0.07 |
| /error-in-render avg req/sec | 1655.35 | 1581.84 | ⚠️ -73.51 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| 450.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.3 kB | 42.3 kB | ✓ |
| main-HASH.js gzip | 30.4 kB | 30.4 kB | ✓ |
| webpack-HASH.js gzip | 1.44 kB | 1.44 kB | ✓ |
| Overall change | 74.3 kB | 74.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.35 kB | 1.35 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 305 B | 305 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.36 kB | 2.36 kB | ✓ |
| head-HASH.js gzip | 342 B | 342 B | ✓ |
| hooks-HASH.js gzip | 906 B | 906 B | ✓ |
| image-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.19 kB | 2.19 kB | ✓ |
| routerDirect..HASH.js gzip | 314 B | 314 B | ✓ |
| script-HASH.js gzip | 375 B | 375 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 14.1 kB | 14.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| _buildManifest.js gzip | 458 B | 458 B | ✓ |
| Overall change | 458 B | 458 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ramosbugs/next.js run-tests-errors | Change |
| - | - | - | - |
| index.html gzip | 530 B | 530 B | ✓ |
| link.html gzip | 545 B | 545 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
Commit: 001cfdbe8d29adc9a285be82d5fadf75bb60cfee

**Pull request merged**